### ISSUE TITLE
Fix optimizer defaults for 558 models flagged by AIDN030

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#992]
## Summary

AIDN030 now uses category-based optimizer intersection to detect models whose category annotations have conflicting optimizer requirements. **558 unique models** are flagged with optimizer conflicts that need to be verified and fixed.

### 1. Models with no optimizer at all
Some models (e.g., `AttentionNetwork`) do manual gradient descent in their `Train()` method by calling `UpdateParameters(GetParameters())` without any optimizer. These models need:
- An `IOptimizer` field with a proper default (Adam/AdamW for transformers, Adam for general neural networks)
- Constructor parameter to allow users to override the optimizer
- The `Train()` method updated to use the optimizer for parameter updates

### 2. Models that inherit a permissive base class
Models that extend `NeuralNetworkBase` may inherit default behavior that doesn't set an optimizer. When these models have restrictive categories like `[Transformer]` or `[GAN]`, the category says "don't use SGD" but the model has no optimizer enforcement.

### 3. Models that correctly use Adam/AdamW already
Many flagged models (e.g., all GAN subclasses, CMGAN, HiFiGAN, BigVGAN) already default to Adam/AdamW. For these, the AIDN030 diagnostic is correctly identifying that their `NeuralNetwork` category allows SGD even though the model doesn't use it. The fix for these is to verify the default is correct (it is) -- no code change needed, but the category combination is still flagged.

## How AIDN030 now works

For each model, the generator:
1. Collects each category's compatible optimizer set separately
2. Computes the intersection across all categories
3. If the intersection is smaller than the union, categories disagree and AIDN030 fires
4. The generated compatibility matrix uses the intersection (safe set)

Example: Model with `[NeuralNetwork, GAN]`
- NeuralNetwork allows: SGD, Adam, AdamW, RMSProp, AdaGrad, Adadelta
- GAN allows: Adam, RMSProp, AdamW
- Intersection: Adam, RMSProp, AdamW (SGD, AdaGrad, Adadelta removed)
- AIDN030 fires because optimizers were removed

### For models without an optimizer (highest priority)
Add an optimizer field with industry-standard default:

```csharp
private readonly IOptimizer<T, Tensor<T>, Tensor<T>> _optimizer;

public MyModel(/* existing params */, IOptimizer<T, Tensor<T>, Tensor<T>>? optimizer = null)
{
    _optimizer = optimizer ?? new AdamOptimizer<T, Tensor<T>, Tensor<T>>(this);
}
```

### For models that already use the correct optimizer
These are false positives from the category intersection. No code change needed -- the model is already correct. To suppress AIDN030 for these, we could add a `[SuppressMessage]` attribute or make the diagnostic smarter by also checking for the presence of an optimizer field.

### Industry-standard optimizer defaults by model family
- **GAN models**: Adam (lr=0.0002, beta1=0.5, beta2=0.999)
- **Transformer models**: AdamW (lr=1e-4 to 3e-4, weight decay=0.01)
- **Diffusion models**: AdamW (lr=1e-4)
- **General neural networks**: Adam (lr=0.001)
- **RNNs/LSTMs**: Adam (lr=0.001) or RMSProp
- **Graph networks**: Adam (lr=0.01)

## Models flagged (558 unique)

These need individual verification. Models already using Adam/AdamW are correct but still flagged due to category intersection.

## Related
- PR #986: Source generators and model type removal
- Issue #990: Comprehensive integration tests

## Full list of 558 flagged models

- ABINet
- ACEStep
- ACGAN
- ALIGN
- ARDiffusionModel
- ARIMAModel
- ARIMAXModel
- ARMAModel
- ARModel
- ASTModel
- AVHuBERT
- AllegroModel
- AquilaVL
- Aria
- AsymmDiTPredictor
- AttentionAllocation
- AttentionNetwork
- AudioFlamingo2
- AudioGenModel
- AudioLDMModel
- AudioLM
- AudioMAE
- AudioPaLM
- AuraFlowModel
- AutoDiffTabGenerator
- AutoIntNetwork
- AutoRegressiveMaskedDiffusion
- Autoformer
- AutoformerModel
- BASIC
- BERTNER
- BESTRQ
- BGE
- BLINKNER
- BLIP3
- BSRoFormer
- BayesianStructuralTimeSeriesModel
- BigGAN
- BigVGAN
- BioBERTNER
- Blip2NeuralNetwork
- BlipNeuralNetwork
- BloombergGPT
- CCDM
- CMGAN
- CSDI
- CTABGANPlusGenerator
- CTGANGenerator
- Cambrian1
- Canary
- CausalGANGenerator
- Chameleon
- Chirp
- Chirp2
- Chirp3
- Chronos
- ChronosBolt
- ChronosFoundationModel
- ClaudeVision
- ClinicalBERTNER
- ClipNeuralNetwork
- CoCa
- CodeBERT
- CodeT5
- CogVLM
- CogVLM2
- CogVideo
- CogVideoX15Model
- CogView4Model
- ColBERT
- Concerto
- ConditionalGAN
- Conformer
- ConformerFP
- CopulaGANGenerator
- CosmosModel
- CosyVoice2
- CoxProportionalHazards
- Crossformer
- CycleGAN
- CycleGANTurboModel
- DCGAN
- DCRNN
- DGCNN
- DIAMONDModel
- DINOX
- DINOv3
- DPCTGANGenerator
- DannaSep
- Data2Vec2
- Data2VecASR
- DeBERTaNER
- DecisionTransformerAgent
- DeepHit
- DeepSeekVL
- DeepSeekVL2
- DeepSurv
- DepthAnythingV2
- Dessurt
- DiT
- DiTNoisePredictor
- DiffusionAutoML
- DistilBERTNER
- DocBank
- DocFormer
- DocGCN
- DocOwl
- DocPedia
- Donut
- Dragonfly
- DragonflyMed
- DualXVSR
- DynamicRegressionWithARIMAErrors
- ELECTRANER
- EMAVFI
- EMMDiTPredictor
- ETSformer
- Eagle
- Eagle25
- Emotion2Vec
- Emu
- Emu2
- Emu3
- EmuVideo2Model
- EmuVideoModel
- ExponentialSmoothingModel
- FEDformer
- FTTransformerNetwork
- FactorTransformer
- FastConformer
- Ferret
- FerretV2
- FinBERT
- FinBERTNER
- FinBERTTone
- FinDiffGenerator
- FinGPT
- FinMA
- FinancialBERT
- FishSpeech
- FlagDiTPredictor
- FlamingoNeuralNetwork
- Florence2
- FlowEditModel
- FlowFormer
- FlowFormerPlusPlus
- Flux1Model
- Flux2Model
- Flux2SchnellModel
- FluxDoubleStreamPredictor
- FluxInpaintingModel
- FluxSchnellModel
- FuseFormer
- GARCHModel
- GLM4Voice
- GLaMM
- GMFlow
- GOGGLEGenerator
- GPT4Point
- GPT4TS
- GR00TN1
- GameGenXModel
- GeminiVision
- Gemma3
- GenerativeAdversarialNetwork
- Genie2Model
- GeoChat
- Gpt4VisionNeuralNetwork
- GraFPrint
- GraphAttentionNetwork
- GraphCodeBERT
- GraphGenerationModel
- GraphIsomorphismNetwork
- GraphNeuralNetwork
- GraphNeuralOperator
- GraphSAGENetwork
- GraphWaveNet
- GrokVision
- Groma
- GroundedSAM2
- GroundingDINO15
- HTDemucs
- Helix
- HiDreamModel
- HiFiGAN
- HuBERT
- HuBERTASR
- HuBERTSER
- HunyuanDiTModel
- HunyuanVideo15Model
- HunyuanVideoModel
- IART
- ICEditModel
- IDEFICS
- IDEFICS2
- IDEFICS3
- ITransformer
- Ideogram3Model
- ImageBindNeuralNetwork
- Imagen3Model
- InfoGAN
- InfographicVQA
- Informer
- InformerModel
- InstructBLIP
- InstructionNER
- InstructorEmbedding
- InternVL
- InternVL2
- InternVL25
- InternVL3
- InternViT
- InternVideo2
- InterventionAnalysisModel
- InvestLM
- JambaLanguageModel
- Janus
- JanusPro
- KOSMOS1
- KOSMOS2
- Kairos
- KaplanMeierEstimator
- KimiVL
- KimiVLThinking
- Kling26Model
- KlingModel
- Kronos
- KyutaiMoshi
- LEOVL
- LISA
- LLM2CLIP
- LLMTime
- LLaVACoT
- LLaVAMed
- LLaVANeXTVideo
- LLaVANeuralNetwork
- LLaVAOneVision
- LLaVAOneVision15
- LLaVAVideo
- LTXVideoModel
- LagLlama
- LatteModel
- LayoutGraph
- LayoutLM
- LayoutLMv2
- LayoutLMv3
- LayoutXLM
- LegalBERTNER
- LiLT
- Llama32Vision
- LlamaOmni
- LogNormalAFT
- LongVILA
- LoongModel
- LumaRay2Model
- LumaRay3Model
- LuminaImage2Model
- LuminaT2XModel
- MAGI1Model
- MAModel
- MARModel
- MATCHA
- MERT
- MGIE
- MGTSD
- MIAVSR
- MMDiTNoisePredictor
- MMDiTXNoisePredictor
- MMS
- MOIRAI
- MOMENT
- MPLUGDocOwl
- MPLUGDocOwl15
- MPLUGDocOwl2
- MPSENet
- MTGNN
- MatchaTTS
- MatryoshkaEmbedding
- MedFlamingo
- MedSAM
- MedSynthGenerator
- MeissonicModel
- MelBandRoFormer
- MiDaS
- MidJourneyV7Model
- MinMo
- MinimaxVideoModel
- MisGANGenerator
- MoMaskModel
- Mochi1Model
- Mochi1PreviewModel
- Moshi
- MotionDiffuseModel
- MotionDiffusionModel
- MovieGenModel
- MusicFlamingo
- MusicGenModel
- MusicTaggingTransformer
- NBEATSModel
- NHiTSModel
- NVLM
- NelsonAalenEstimator
- NeuralGARCH
- NeuralNetworkARIMAModel
- NeuralProgramSynthesizer
- NonStationaryTransformer
- Nougat
- OCTGANGenerator
- OMGLLaVA
- ONNXNER
- OasisModel
- Octo
- OmniGen2
- OmniGen2Model
- OmniGenModel
- OmnilangualASR
- OneFormer
- OpenFlamingo
- OpenSora
- OpenSora13Model
- OpenSora2Model
- OpenSoraModel
- OpenVocabSAM
- PICK
- PLLaVA
- PSRT
- PURENER
- PaLI
- PaLI3
- PaLIX
- PaLME
- PatchTST
- PathVLM
- Pengi
- PerceptionEncoder
- Phi4Multimodal
- PiZero
- Pika21Model
- Pix2Pix
- Pix2Struct
- PixArtDeltaLCMModel
- PixArtDeltaModel
- PixArtModel
- PixArtSigmaModel
- PixtralLarge
- PlaygroundV3Model
- PointEModel
- PointLLM
- ProPainter
- ProgressiveGAN
- PromptNER
- ProphetModel
- PubMedBERTNER
- PyramidFlowModel
- QVQ72B
- Qwen25VL
- Qwen2Audio
- Qwen2VL
- Qwen3ASR
- Qwen3VL
- RADIOv25
- RELNER
- REaLTabFormerGenerator
- RSGPT
- RT2
- RVRT
- RadFM
- RealESRGANModel
- RealViformer
- RealizedVolatilityTransformer
- RecraftV3Model
- RoBERTaNER
- RunwayGen4Model
- SAGAN
- SAINT
- SAINTNetwork
- SALMONN
- SAM
- SAM2
- SAM21
- SAMHQ
- SANAModel
- SANASprintModel
- SARIMAModel
- SD3FlashModel
- SD3InpaintingModel
- SD3TurboModel
- SDTurboModel
- SDXLLightningModel
- SDXLTurboModel
- SECBERT
- SECBertNER
- SEEDX
- SEEM
- SGPT
- SPIRAL
- SPLADE
- STGNN
- STLDecomposition
- STTN
- SVTR
- SambaLanguageModel
- SceneLLM
- SciBERTNER
- SeedASR
- SeedEdit3Model
- SeedVR
- Seedance1Model
- SegGPT
- SenseFlowModel
- ShapEModel
- Shikra
- ShowO
- ShowO2
- SiDDiTModel
- SiTPredictor
- SimCSE
- SimMTM
- SkyEyeGPT
- SkyReelsV1Model
- SkyworkR1V
- SkyworkR1V2
- SlowFastLLaVA
- SmartEdit
- SnapVideoModel
- Sora2Model
- SoraModel
- SpERTNER
- SpanBERTNER
- SpeakerLM
- SpectralAnalysisModel
- SpeechGPT
- SpiralNet
- SpiritLM
- StableAudioModel
- StableDiffusion35Model
- StableDiffusion3Model
- StateSpaceModel
- Step1XEditModel
- StepAudio
- StepVideoModel
- StyleGAN
- StyleTTS2
- Sundial
- SwinVFI
- TBATSModel
- TEST
- TFGridNet
- TFT
- TLBVFI
- TOTO
- TRIE
- TSDiff
- TSMixer
- TTVSR
- TVAEGenerator
- TabDDPMGenerator
- TabDPTNetwork
- TabFlowGenerator
- TabLLMGenGenerator
- TabPFNNetwork
- TabSynGenerator
- TabTransformer
- TabTransformerGenGenerator
- TabTransformerNetwork
- TableGANGenerator
- TableTransformer
- TemplateNER
- TemporalFusionTransformer
- TemporalGCN
- TextMonkey
- ThreeDGraphLLM
- ThreeDLLM
- ThreeDVLA
- TimeBridge
- TimeDiff
- TimeGANGenerator
- TimeGPT
- TimeGrad
- TimeLLM
- TimeMAE
- TimeMoE
- TimeSeriesRegression
- TimeSformer
- Timer
- TimesFM
- TinyBERTNER
- TrOCR
- TransUNet
- TransferFunctionModel
- Transformer
- TransformerEmbeddingNetwork
- Transfusion
- TransfusionModel
- TriaffineNER
- TripoSRModel
- TurboEditModel
- TurboFillModel
- UDOP
- UFM
- UNetNoisePredictor
- UReader
- USM
- UViTNoisePredictor
- UniAudio
- UniMatch
- UniSimModel
- UniSpeech
- UniTS
- UnifiedMultimodalNetwork
- UnobservedComponentsModel
- VALLE
- VARMAModel
- VFIT
- VFIformer
- VITSModel
- VRT
- VectorAutoRegressionModel
- Veo3Model
- VeoModel
- ViTCoMer
- VideoCLIP
- VideoCLIPNeuralNetwork
- VideoChat2
- VideoLISA
- VideoLLaMA2
- VideoLLaMA3
- VideoLLaVA
- VideoMAE
- VideoPoetModel
- VideoUNetPredictor
- VisionTS
- VisionTransformer
- VoiceCraft
- W2NER
- W2vBERT
- WGAN
- WGANGP
- Wan21Model
- Wan22Model
- WanVideoModel
- Wav2Vec2
- Wav2Vec2ASR
- Wav2Vec2LanguageIdentifier
- Wav2Vec2Model
- WavLM
- WavLMASR
- WavLMRobust
- WavLMSpeaker
- WeibullAFT
- WhisperModel
- XDecoder
- XLMRoBERTaNER
- XLSR
- YingLong
- YuE
- Zamba2LanguageModel
- ZambaLanguageModel
